### PR TITLE
fix: decode token payload with utf-8 characters

### PR DESCRIPTION
**Pull request by ofoucherot**
Fix an issue when token payload contains user info with accented characters and were not properly decoded.

### CODE REVIEW

File: packages/liveblocks-client/src/utils.ts [hunk 1]
```diff
@@ -431,3 +431,24 @@ export function values<O extends { [key: string]: unknown }>(
 ): O[keyof O][] {
   return Object.values(obj) as O[keyof O][];
 }
+
+/**
+ * Decode JWT Token payload part. Properly decode utf-8 characters.
+ */
+export function decodeJwtTokenPayload(payload: string): Json | undefined {
```
[the remark this answers is not in the document]
Comment (ofoucherot) [reply packages/liveblocks-client/src/utils.ts]: Yes, it makes sense!
I'm ending with this `tryParseJson(b64decode(tokenParts[1]));`.

[the remark this answers is not in the document]
Comment (nvie) [reply packages/liveblocks-client/src/utils.ts]: Sweet!

File: packages/liveblocks-client/src/utils.ts [hunk 2]
```diff
@@ -431,3 +431,37 @@ export function values<O extends { [key: string]: unknown }>(
...
+        .split("")
+        .map(function (c) {
+          return "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2);
+        })
+        .join("")
+    );
+
+    return decodedValue;
+  } catch (err) {
+    return atob(b64value);
```
[the remark this answers is not in the document]
Comment (ofoucherot) [reply packages/liveblocks-client/src/utils.ts:465]: We still want to default to `atob` call only if the previous code fails. It is inspired by some jwt libraries.

[the remark this answers is not in the document]
Comment (nvie) [reply packages/liveblocks-client/src/utils.ts:465]: Not sure if I understand how the algorithm here works, but if something fails here, wouldn't you possibly return garbage (i.e. accented char bugs) if you try to fall back to the non-utf8 way in the catch clause? If something fails, shouldn't that be reflected in returning an `undefined`?

**Review by nvie — APPROVED** [review]

File: packages/liveblocks-client/src/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ofoucherot) [reply packages/liveblocks-client/src/utils.ts:465]: It is better to return a valid payload even if there are issues with accented char in the user info.

**Comment by nvie** [comment]
Getting this merged now to backport and roll a new 0.16 release today.

**Pull request merged**